### DISCUSSION TITLE
Need help understanding the usage of ssr-profile.js in Auth0 sample

### DISCUSSION THREAD

**Discussion by thesobercoder** [vercel/next.js#15169] [Help] [answered]
I was going through the Auth0 sample and I came across the [`ssr-profile.js`](https://github.com/vercel/next.js/blob/canary/examples/auth0/pages/advanced/ssr-profile.js) file. Trying to make sense of what this file does as I don't see how an user would even arrive at this URL. Is this some special directory thing?

**Comment by juanbzz** [comment, marked as answer]
@thesobercoder that file is an example on how'd you do authorization on a protected page using SSR. 

For example, when you don't want the page to load at all, you [redirect the unauthorized user from the server using `getServerSideProps`](https://github.com/vercel/next.js/blob/canary/examples/auth0/pages/advanced/ssr-profile.js#L26-L32) when the session is invalid.
